crates/evm/core/src/tempo.rs

```diff
@@ -0,0 +1,139 @@
+use std::collections::HashMap;
+
+use alloy_primitives::{Address, U256};
+use revm::{
+    Database,
+    state::{AccountInfo, Bytecode},
+};
+use tempo_chainspec::hardfork::TempoHardfork;
+use tempo_precompiles::{error::TempoPrecompileError, storage::PrecompileStorageProvider};
+
+use crate::backend::Backend;
+
+/// Storage provider adapter for Foundry's backend to work with Tempo precompiles.
+///
+/// This wraps Foundry's backend to implement the `PrecompileStorageProvider` trait,
+/// enabling use of canonical Tempo initialization logic.
+pub struct FoundryStorageProvider<'a> {
```

Comment:
this seems fine and we can just upstream as is i guess. however i think we should look at a more future proof way to do any kind of "genesis" init for different chains